var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -77,18 +80,49 @@ class Netcdf(AutotoolsPackage):
 
     depends_on("m4", type='build')
     depends_on("hdf", when='+hdf4')
+
+    # curl 7.18.0 or later is required:
+    # http://www.unidata.ucar.edu/software/netcdf/docs/getting_and_building_netcdf.html
     depends_on("curl@7.18.0:", when='+dap')
     depends_on("curl@7.18.0:", when='+cdmremote')
-    depends_on('parallel-netcdf', when='@4.2.1.1:+parallel-netcdf')
 
-    # Required for NetCDF-4 support
+    depends_on('parallel-netcdf', when='+pnetcdf')
+
+    # We need to build with MPI wrappers if any of the two
+    # parallel I/O features is enabled:
+    # http://www.unidata.ucar.edu/software/netcdf/docs/getting_and_building_netcdf.html#build_parallel
+    depends_on('mpi', when='+parallel4')
+    depends_on('mpi', when='+pnetcdf')
+
+    # zlib 1.2.5 or later is required for netCDF-4 compression:
+    # http://www.unidata.ucar.edu/software/netcdf/docs/getting_and_building_netcdf.html
     depends_on("zlib@1.2.5:")
-    depends_on('hdf5')
+
+    # High-level API of HDF5 1.8.9 or later is required for netCDF-4 support:
+    # http://www.unidata.ucar.edu/software/netcdf/docs/getting_and_building_netcdf.html
+    # TODO: change to 'hdf5+hl' when/if https://github.com/LLNL/spack/pull/5790
```

Comment:
when/if is merged ?

Reply:
There is no 'hl' variant for hdf5 in the develop branch. The referenced PR introduces it.

Reply:
i know, i am saying you miss `is merged` in the sentence.

Reply:
Ok, done.